app/controllers/thredded/posts_controller.rb

```diff
@@ -53,10 +54,22 @@ def destroy
                     notice: I18n.t('thredded.posts.deleted_notice')
     end
 
+    def mark_as_read
+      authorize post, :read?
+      UserTopicReadState.touch!(thredded_current_user.id, post)
+      respond_to do |format|
+        format.html { redirect_to request.referer || post_path(post, user: thredded_current_user) }
```

Comment:
Please use:

```ruby
redirect_back fallback_location: post_path(post, user: thredded_current_user)
```

We have `redirect_back` polyfilled for Rails 4 here:

https://github.com/thredded/thredded/blob/235d0a9026c9f8248297c0fca13e655aca893a75/app/controllers/thredded/application_controller.rb#L66-L73